oem/ibm/libpldmresponder/file_io_type_chaap.cpp

```diff
@@ -0,0 +1,36 @@
+#include "file_io_type_chaap.hpp"
+
+#include <iostream>
+
+namespace pldm
+{
+using namespace utils;
+
+namespace responder
+{
+
+static constexpr auto chapDataFilePath = "/var/lib/pldm/ChapData/";
+static constexpr auto chapDataFilename = "chapsecret";
+
+int ChapHandler::readIntoMemory(uint32_t offset, uint32_t& length,
+                                uint64_t address,
+                                oem_platform::Handler* /*oemPlatformHandler*/)
+{
+    namespace fs = std::filesystem;
+    if (!fs::exists(chapDataFilePath))
+    {
+        error("chap file directory not present.");
+        return PLDM_ERROR;
+    }
+    std::string filePath = std::string(chapDataFilePath) +
+                           std::string(chapDataFilename);
+    auto rc = transferFileData(filePath.c_str(), true, offset, length, address);
+    fs::remove(filePath);
+    if (rc)
```

Comment:
Below code can be replaced with "return rc"

if (rc)
    {
        return PLDM_ERROR;
    }
    return PLDM_SUCCESS;

Reply:
can not return rc value since we need to return specific error/success pldm code and readability is not good if we remove only rc value.